app/services/targets/generate_progress_service.rb

```diff
@@ -82,6 +78,20 @@ def has_target_for_fuel_type?(fuel_type)
       end
     end
 
+    def target
+      @school.most_recent_target
+    end
+
+    def fetch_latest_figures(hash_of_months_to_values)
+      val = hash_of_months_to_values[reporting_month]
+      #sometimes the schools meter data may be lagging only a few days or a week
+      #behind. this means that the progress report does not have data for this month,
+      #it only has data for the previous month. So if there's no entry for the reporting
+      #month, look for earlier data. This typically only happens around the beginning of
+      #month when we're running a little behind on data
+      val.present? ? val : hash_of_months_to_values[reporting_month.prev_month]
```

Comment:
I might be wrong here but maybe lines 86 and 92 could be replaced with
`hash_of_months_to_values[reporting_month] || hash_of_months_to_values[reporting_month.prev_month]`
?